primo/data_parser/tests/test_metric_data.py

```diff
@@ -0,0 +1,400 @@
+    assert im_wt.dac_impact.submetrics == {
+        "fed_dac": im_wt.fed_dac,
+        "state_dac": im_wt.state_dac,
+    }
+
+    assert hasattr(im_wt, "hospitals")
+    assert hasattr(im_wt, "schools")
+    assert hasattr(im_wt, "buildings_near")
+    assert hasattr(im_wt, "buildings_far")
+    assert im_wt.sensitive_receptors.submetrics == {
+        "hospitals": im_wt.hospitals,
+        "schools": im_wt.schools,
+        "buildings_near": im_wt.buildings_near,
+        "buildings_far": im_wt.buildings_far,
+    }
+
+    assert hasattr(im_wt, "fed_wetlands_near")
+    assert hasattr(im_wt, "fed_wetlands_far")
+    assert hasattr(im_wt, "state_wetlands_near")
+    assert hasattr(im_wt, "state_wetlands_far")
+    assert im_wt.environment.submetrics == {
+        "fed_wetlands_near": im_wt.fed_wetlands_near,
+        "state_wetlands_near": im_wt.state_wetlands_near,
+        "fed_wetlands_far": im_wt.fed_wetlands_far,
+        "state_wetlands_far": im_wt.state_wetlands_far,
+    }
+
+    assert hasattr(im_wt, "ann_production_volume")
+    assert hasattr(im_wt, "five_year_production_volume")
+    assert im_wt.production_volume.submetrics == {
+        "ann_production_volume": im_wt.ann_production_volume,
+        "five_year_production_volume": im_wt.five_year_production_volume,
+    }
+
+    assert hasattr(im_wt, "h2s_leak")
+    assert hasattr(im_wt, "brine_leak")
+    assert im_wt.other_emissions.submetrics == {
+        "h2s_leak": im_wt.h2s_leak,
+        "brine_leak": im_wt.brine_leak,
+    }
+
+    assert not hasattr(im_wt.well_age, "submetrics")
+    assert not hasattr(im_wt.well_count, "submetrics")
+    assert not hasattr(im_wt.well_integrity, "submetrics")
+
+    im_wt.set_value(
+        primary_metrics={
+            "ch4_emissions": 25,
+            "sensitive_receptors": 30,
+            "environment": 20,
+            "dac_impact": 25,
+        },
+        submetrics={
+            "ch4_emissions": {
+                "leak": 30,
+                "compliance": 20,
+                "violation": 50,
+            },
+            "dac_impact": {
+                "fed_dac": 60,
+                "state_dac": 40,
+            },
+            "environment": {
+                "fed_wetlands_near": 50,
+                "state_wetlands_near": 50,
+            },
+            "sensitive_receptors": {
+                "buildings_near": 20,
+                "hospitals": 40,
+                "schools": 40,
+            },
+        },
+    )
+
+    assert im_wt.check_validity() is None
+
+
+def test_efficiency_metrics_class():
+    ef_wt = EfficiencyMetrics()
+
+    assert hasattr(ef_wt, "num_wells")
+    assert hasattr(ef_wt, "num_unique_owners")
+    assert hasattr(ef_wt, "dist_centroid")
+    assert hasattr(ef_wt, "avg_dist_road")
+    assert hasattr(ef_wt, "elevation_delta")
+    assert hasattr(ef_wt, "age_range")
+    assert hasattr(ef_wt, "avg_age")
+    assert hasattr(ef_wt, "depth_range")
+    assert hasattr(ef_wt, "avg_depth")
+
```

Comment:
Same here. We will need to consider "record completeness" as an efficiency term to test for or add a test for it in the next update.